### PR TITLE
Eq 410 navigator

### PR DESCRIPTION
**Pull request by warrenbailey**
### What is the context of this PR?

Removing the navigator and replacing it with the User Journey Manager. Also cleaning up the User action processor and moving the routing components into the routing engine. 
### How to review
1. Check out this branch
2. Check the application still works
3. Ensure all tests pass
- [ ] Do all commits have sensible commit messages?
- [ ] Is all new code unit tested?
- [ ] Are there integration tests if needed?
- [ ] Is there sufficient logging?
- [ ] Is there documentation or is the code self-describing?
### Who worked on the PR

@warrenbailey

### CODE REVIEW

File: app/questionnaire_state/user_journey_manager.py [hunk 1]
```diff
@@ -63,22 +77,33 @@ def go_to_state(self, item_id):
             self._create_new_state(item_id)
 
     def _create_new_state(self, item_id):
-        item = self._schema.get_item_by_id(item_id)
+
         logger.debug("Creating new state for %s", item_id)
-        if isinstance(item, SchemaBlock):
-            state = StateBlock.construct_state(item)
+        if item_id in self._valid_locations:
```
Comment (dhilton) [inline app/questionnaire_state/user_journey_manager.py:80]: I wonder if there is any way we can remove this case style statement and use polymorphism? 
http://refactoring.com/catalog/replaceConditionalWithPolymorphism.html

File: app/routing/routing_engine.py [hunk 2]
```diff
@@ -1,10 +1,40 @@
...
     ''' The routing engine will apply any routing rules dependant on where the user is in the schema
     and what answers they have provided.
     '''
     def __init__(self, schema_model):
         self._schema = schema_model
 
-    def get_next(self, current_location):
-        # however we're yet to implement routing so always return the current location
-        return current_location
+    def get_next_location(self, current_location):
```
Comment (dhilton) [inline app/routing/routing_engine.py:12]: When we talk about the next location, we really mean the next block to display? Would it ever be anything else?

Comment (warrenbailey) [inline app/routing/routing_engine.py:12]: Technically no, but the system is flexible enough to cope with us changing it to question or section for example. That's why we haven't called it get_next_block

File: app/questionnaire_state/user_journey_manager.py (hunk 1 above)
Comment (warrenbailey) [inline app/questionnaire_state/user_journey_manager.py:80]: That doesn't really work in this case as item isn't a StateBlock its SchemaBlock. We could modify the schema model objects to have a construct state method, which could work.

**Comment by dhilton** [comment]
Happy to merge this now?

**Comment by warrenbailey** [comment]
Yep am now

**Pull request merged**